vassal-app/src/main/java/VASSAL/build/module/documentation/BrowserPDFFile.java

```diff
@@ -0,0 +1,196 @@
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License (LGPL) as published by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, copies are available
+ * at http://www.opensource.org.
+ */
+package VASSAL.build.module.documentation;
+
+import VASSAL.Info;
+import VASSAL.build.AbstractConfigurable;
+import VASSAL.build.Buildable;
+import VASSAL.build.GameModule;
+import VASSAL.i18n.Resources;
+import VASSAL.tools.BrowserSupport;
+import VASSAL.tools.DataArchive;
+import VASSAL.tools.menu.MenuItemProxy;
+import VASSAL.tools.menu.MenuManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.swing.AbstractAction;
+import javax.swing.Action;
+import java.awt.event.ActionEvent;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+
+/**
+ * Easier-to-use support for opening a single PDF file in the browser (it can be done with BrowserHelpFile, but
+ * requires a substantial workaround and weird confusing procedures)
+ */
+public class BrowserPDFFile extends AbstractConfigurable {
+  private static final Logger logger =
+    LoggerFactory.getLogger(BrowserPDFFile.class);
+
+  public static final String TITLE = "title"; //$NON-NLS-1$
+  public static final String PDF_FILE = "pdfFile"; //$NON-NLS-1$
+  protected String menuText = Resources.getString("Editor.BrowserPDFFile.default_menu_text");
+  protected String pdfFile = Resources.getString("Editor.BrowserPDFFile.default_filename");
+  protected Action launch;
+  protected URL url;
+
+  public BrowserPDFFile() {
+    super();
+
+    launch = new AbstractAction() {
+      private static final long serialVersionUID = 1L;
+
+      @Override
+      public void actionPerformed(ActionEvent e) {
+        launch();
+      }
+    };
+  }
+
+  public void launch() {
+    if (url == null) {
+      extractPDF();
+    }
+    if (url != null) {
+      BrowserSupport.openURL(url.toString());
+    }
+  }
+
+  /**
+   * Extracts our PDF to a temporary file in the temp directory
+   */
+  protected void extractPDF() {
+    final DataArchive archive = GameModule.getGameModule().getDataArchive();
+
+    try (InputStream in = archive.getInputStream(pdfFile)) {
+      Path out = Files.createTempFile(Info.getTempDir().toPath(), "pdfhelp_", ".pdf"); //NON-NLS
+      try {
+        Files.copy(in, out, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
You should add an `out.toFile().deleteOnExit();` here so that the file gets cleaned up.

Reply:
done